test/test_nestedtensor.py

```diff
@@ -8654,6 +8654,111 @@ def test_with_factor(self):
         self.assertTrue(a * 2 == 2 * a)
 
 
+from torch.nested._internal.cached_tensor import CachedTensor
```

Comment:
I would agree with you haha, sadly it's hard to behave like a true CachedTensor without metadata mutation, so  suggestions welcome/happy to bikeshed some more.

Also given the approach of handling the source + extra fields distinction at a different layer, it could even be something like `ContainerTensor`.